docker/entrypoint.sh

```diff
@@ -0,0 +1,67 @@
+#!/bin/sh
+
+# Define the start command
+START_COMMAND="teosd"
+
+# Set the API bind address
+if [[ ! -z ${API_BIND} ]]; then
+    START_COMMAND="$START_COMMAND --apibind $API_BIND"
+fi
+
+# Set the API port
+if [[ ! -z ${API_PORT} ]]; then
+    START_COMMAND="$START_COMMAND --apiport $API_PORT"
+fi
+
+# Set the RPC bind address
+if [[ ! -z ${RPC_BIND} ]]; then
+    START_COMMAND="$START_COMMAND --rpcbind $RPC_BIND"
+fi
+
+# Set the RPC port
+if [[ ! -z ${RPC_PORT} ]]; then
+    START_COMMAND="$START_COMMAND --rpcport $RPC_PORT"
+fi
+
+# Set the Bitcoin network
+if [[ ! -z ${BTC_NETWORK} ]]; then
+    START_COMMAND="$START_COMMAND --btcnetwork $BTC_NETWORK"
+fi
+
+# Set the Bitcoin RPC credentials
+if [[ ! -z ${BTC_RPC_USER} ]]; then
+    START_COMMAND="$START_COMMAND --btcrpcuser $BTC_RPC_USER"
+fi
+
+if [[ ! -z ${BTC_RPC_PASSWORD} ]]; then
+    START_COMMAND="$START_COMMAND --btcrpcpassword $BTC_RPC_PASSWORD"
+fi
+
+# Set the Bitcoin RPC connection details
+if [[ ! -z ${BTC_RPC_CONNECT} ]]; then
+    START_COMMAND="$START_COMMAND --btcrpcconnect $BTC_RPC_CONNECT"
+fi
+
+if [[ ! -z ${BTC_RPC_PORT} ]]; then
+    START_COMMAND="$START_COMMAND --btcrpcport $BTC_RPC_PORT"
+fi
+
+if [[ ! -z ${DEBUG} ]]; then
+    START_COMMAND="$START_COMMAND --debug"
+fi
+
+if [[ ! -z ${DEPS_DEBUG} ]]; then
+    START_COMMAND="$START_COMMAND --depsdebug"
+fi
+
+if [[ ! -z ${OVERWRITE_KEY} ]]; then
+    START_COMMAND="$START_COMMAND --overwritekey"
+fi
+
+if [[ ! -z ${FORCE_UPDATE} ]]; then
+    START_COMMAND="$START_COMMAND --forceupdate"
+fi
```

Comment:
These four don't need to be set to some string, thus we should use `! -z ${VARIABLE+x}` which checks if the variable is set even if it is an empty string.

Reply:
Hmmm I worry slightly about this because if someone is used to setting the variables as true or false in the config variable might be confused when they set one of the values above to "false" but this logic considers this to be "true"

But I definitely see what you're saying. What do you think about doing something like this: setting the above as boolean strings (since in shell script booleans are... confusing). We write a comment noting that to turn them on they must be set the string to "true". And then we change the above logic to just `if [ "$DEBUG" = true ]; then` or something along those lines.

Reply:
Agreed!

Reply:
Good catch. I made those updates, let me know what you think